### ISSUE TITLE
[DocDB] Intent RecordStats counters are slowing down scans

### ISSUE THREAD

**Issue by hari90** [yugabyte/yugabyte-db#21832]
Jira Link: [DB-10734](https://yugabyte.atlassian.net/browse/DB-10734)
### Description

```
│W0331 18:18:57.147332 59519 long_operation_tracker.cc:114] Read running for 6.027s in thread 95130:                                       │
│    @          0x2a253d9  yb::AtomicGauge<>::IncrementBy()                                                                                │
│    @          0x33a882b  rocksdb::StatisticsMetricImpl::recordTick()                                                                     │
│    @          0x32c73a5  rocksdb::DBIter::Next()                                                                                         │
│    @          0x2f2a305  yb::docdb::PerformRocksDBSeek()                                                                                 │
│    @          0x2f331c6  yb::docdb::IntentAwareIterator::Seek()                                                                          │
│    @          0x2ef16aa  yb::docdb::HybridScanChoices::SeekToCurrentTarget()                                                             │
│    @          0x2ef68be  yb::docdb::DocRowwiseIterator::HasNext()                                                                        │
│    @          0x34f41b6  yb::tablet::AbstractTablet::HandleQLReadRequest()                                                               ┤
│    @          0x35331e2  yb::tablet::Tablet::HandleQLReadRequest()                                                                       │
│    @          0x36c8cc3  yb::tserver::(anonymous namespace)::ReadQuery::Complete()                                                       │
│    @          0x36c5b66  yb::tserver::PerformRead()                                                                                      ┤
│    @          0x3705601  yb::tserver::TabletServiceImpl::Read()                                                                          ┤
│    @          0x37dde33  std::__1::__function::__func<>::operator()()                                                                    ┤
│    @          0x37df695  yb::tserver::TabletServerServiceIf::Handle()                                                                    │
│    @          0x348bd87  yb::rpc::ServicePoolImpl::Handle()                                                                              ┤
│    @          0x33da384  yb::rpc::InboundCall::InboundCallTask::Run()                                                                    │

```

161 out of 313 collected stack traces point to the metric

### Issue Type

kind/bug

### Warning: Please confirm that this issue does not contain any sensitive information

- [X] I confirm this issue does not contain any sensitive information.

[DB-10734]: https://yugabyte.atlassian.net/browse/DB-10734?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by bmatican** [comment]
@hari90 is this in 2.14, or latest? @es1024 was your per-RPC metrics work in 2.14?

**Comment by rthallamko3** [comment]
@bmatican , This was from 2.14.x. So with @es1024 's changes for per-RPC metrics tracking, this might not be an issue anymore.

**Comment by es1024** [comment]
@bmatican 2.14 does not have per-RPC metric updates; the stack does look like the issue that I fixed for 2.18+.
